app/api/v1/vaults.py

```
@@ -51,12 +65,50 @@ def provision_vault(
+    vault_id: str,
+    uc: SendUnlockCommand = Depends(get_send_unlock_command_uc),
+    current_user: User = Depends(get_current_user),
+) -> UnlockCommandResponse:
+    try:
+        result = await uc.execute(vault_id=vault_id, user_id=current_user.id)
+    except UnauthorizedVaultAccessError as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
+    except InsufficientPermissionsError as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
+    except ValueError:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Vault not found")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)